net/socks5/init.go

```diff
@@ -0,0 +1,34 @@
+package socks5
+
+import "fmt"
+
+const MaxInitRequestSize = 257
+
+func MethodsFromInitPacket(pkt []byte) ([]AuthMethod, error) {
```

Comment:
This would be a great candidate for fuzzing. Ping me if you'd like to pair on doing that.